### DISCUSSION TITLE
Is there available cli command for pulling environment variable for specific branch and also deploying to specific branch

### DISCUSSION THREAD

**Discussion by ezalorsara** [vercel/community#943] [Help] [answered]
Is there available cli command for pull environment variable for specific branch and also deploying to specific branch?
I only find adding environment variables to specific branch
vercel env add [name] [environment] [gitbranch] < [file]

Thanks

**Comment by amyegan** [comment]
Thanks for reaching out @ezalorsara! You can use `vercel env ls [environment] [gitbranch]` to get a list of environment variables for a specific branch. There is no way to specify a branch on `vercel pull` yet, but the team is looking at adding that option.

**Reply by ezalorsara** [reply]
Hi @amyegan  please also add features to deploy on a specific branch, it would be very helpful.

The idea is  i have this cicd that will automatically create database for specific branch, let say there is a feature/accounting branch it will have its own database.

Steps:
1.) initialize database to be used in a specific branch feature/accounting
2.) store the database url generated on previous step to vercel environment for the branch feature/accounting
3.) deploy vercel to specific branch.

it also would be great if we can delete environment variables on specific branch via vercel cli so that i don't need to use the vercel rest api.

Thank you hope you consider.

**Comment by dshunfen** [comment]
+1 to this! I went in thinking this was expected behavior for `vercel pull`. Would be great to have that feature.
I'm doing CI builds in GH actions, and would benefit from building it in GH rather than on vercel itself (which is slower and takes much longer to download all necessary dependencies).

**Comment by amyegan** [comment, marked as answer]
Hi @ezalorsara and @dshunfen, I have some good news to share! The CLI team just added a `--git-branch` option to `vc pull` and `vc env pull` in [vercel@28.9.0](https://github.com/vercel/vercel/releases/tag/vercel%4028.9.0).
